src/components/SearchBar/SearchBar.js

```diff
@@ -18,11 +19,14 @@ const SearchBar = () => {
         type="text"
         placeholder="Search product..."
         onChange={(event) => {
-          var text = event.target.value;
-          text.length < 3 ||
-            dispatch(updateFilter({ filter: text, type: 'text' }));
-          text.length ||
-            dispatch(updateFilter({ filter: 0, type: 'category' }));
+          clearTimeout(timeout);
```

Comment:
I'm not sure about this. It's working good. It will wait one second before to make a request to avoid a "too many request" response. What do you think @arieltapia16? Do you know a better way to resolve this?